### DISCUSSION TITLE
Slight clarification on threads

### DISCUSSION THREAD

**Discussion by brc-dd** [vitest-dev/vitest#3057] [Q&A]
We have certain tests in VitePress that are too much flaky. Debugging them, we reached to conclusion that certain tests are overlapping. For example, there are tests for the `vitepress init` command, where we clean a directory, scaffold a template variation, run tests, clean it up and repeat this for all template variations. But this is randomly failing as sometimes that directory is cleaned before the tests can finish. Can this be due to threading (should we set singleThread true) or should we be looking at something else?

**Comment by sheremet-va** [comment]
Vitest runs several test files at once, so it is possible to have an overlap, yes. You can enable `--single-thread` or use `poolMatchGlobs` to run these tests in a `child_process`, because it's always sequential.

**Reply by brc-dd** [reply]
Thanks for quick reply 🙌  But still no luck. IG something else is the issue. Here is the file that fails: https://github.com/vuejs/vitepress/blob/main/__tests__/init/init.test.ts, does anything look "obviously wrong" there 👀 ?
